executor/api/rest/server.go

```diff
@@ -0,0 +1,254 @@
+	if !probesOnly {
+		serverMetrics = metric.NewServerMetrics(predictor, deploymentName)
+	}
+	return &SeldonRestApi{
+		mux.NewRouter(),
+		client,
+		predictor,
+		logf.Log.WithName("SeldonRestApi"),
+		probesOnly,
+		serverUrl,
+		namespace,
+		protocol,
+		deploymentName,
+		serverMetrics,
+		prometheusPath,
+	}
+}
+
+func (r *SeldonRestApi) respondWithSuccess(w http.ResponseWriter, code int, payload payload.SeldonPayload) {
+	w.Header().Set("Content-Type", payload.GetContentType())
+	w.WriteHeader(code)
+
+	err := r.Client.Marshall(w, payload)
+	if err != nil {
+		r.Log.Error(err, "Failed to write response")
+	}
+}
+
+func (r *SeldonRestApi) respondWithError(w http.ResponseWriter, err error) {
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusInternalServerError)
+
+	errPayload := r.Client.CreateErrorPayload(err)
+	err = r.Client.Marshall(w, errPayload)
+	if err != nil {
+		r.Log.Error(err, "Failed to write error payload")
+	}
+}
+
+func (r *SeldonRestApi) wrapMetrics(service string, baseHandler http.HandlerFunc) http.HandlerFunc {
+
+	handler := promhttp.InstrumentHandlerDuration(
+		r.metrics.ServerHandledHistogram.MustCurryWith(prometheus.Labels{
+			metric.DeploymentNameMetric:   r.DeploymentName,
+			metric.PredictorNameMetric:    r.predictor.Name,
+			metric.PredictorVersionMetric: r.predictor.Annotations["version"],
+			metric.ServiceMetric:          service}),
+		baseHandler,
+	)
+	return handler
+}
+
+func (r *SeldonRestApi) Initialise() {
+	r.Router.HandleFunc("/ready", r.checkReady)
+	r.Router.HandleFunc("/live", r.alive)
+	r.Router.Handle(r.prometheusPath, promhttp.Handler())
+	if !r.ProbesOnly {
+		r.Router.Use(puidHeader)
+		switch r.Protocol {
+		case api.ProtocolSeldon:
+			//v0.1 API
+			api01 := r.Router.PathPrefix("/api/v0.1").Methods("POST").Subrouter()
+			api01.Handle("/predictions", r.wrapMetrics(metric.PredictionHttpServiceName, r.predictions))
+			r.Router.NewRoute().Path("/api/v0.1/status/{" + ModelHttpPathVariable + "}").Methods("GET").HandlerFunc(r.wrapMetrics(metric.StatusHttpServiceName, r.status))
+			r.Router.NewRoute().Path("/api/v0.1/metadata/{" + ModelHttpPathVariable + "}").Methods("GET").HandlerFunc(r.wrapMetrics(metric.StatusHttpServiceName, r.metadata))
+			//v1.0 API
+			api1 := r.Router.PathPrefix("/api/v1.0").Methods("POST").Subrouter()
+			api1.Handle("/predictions", r.wrapMetrics(metric.PredictionServiceMetricName, r.predictions))
+			r.Router.NewRoute().Path("/api/v1.0/status/{" + ModelHttpPathVariable + "}").Methods("GET").HandlerFunc(r.wrapMetrics(metric.StatusHttpServiceName, r.status))
+			r.Router.NewRoute().Path("/api/v1.0/metadata/{" + ModelHttpPathVariable + "}").Methods("GET").HandlerFunc(r.wrapMetrics(metric.StatusHttpServiceName, r.metadata))
+
+		case api.ProtocolTensorflow:
+			r.Router.NewRoute().Path("/v1/models/{" + ModelHttpPathVariable + "}/:predict").Methods("POST").HandlerFunc(r.wrapMetrics(metric.PredictionHttpServiceName, r.predictions))
```

Comment:
I can also see that if tensorflow protocol is enabled, then the graph is restricted to tensorflow API only for all components in the graph, which means you would also not be able to use Python wrapper components... it would be good to discuss further, as also the limitation of having only components of either GRPC or REST (and not combination of both) would be an issue for the model that we built in the project last year

Reply:
Maybe need customer feedback on paths unification

Reply:
Need to get feedback if we want to support mixed protocols in graph